app/Controller/EventsController.php

```diff
@@ -2802,9 +2802,15 @@ public function freeTextImport($id) {
 		}
 
 		if ($this->request->is('post')) {
+            //getting the warninglists
+            App::uses('WarninglistsController', 'Controller');
+            $warningList = new WarninglistsController();
```

Comment:
You should never have to call functions from a different controller, this breaks basic MVC principles.